### PR TITLE
Start populating the IDT and handle double faults.

### PR DESCRIPTION
**Pull request by andrisaar**

### CODE REVIEW

**Review by conradgrobler — APPROVED** [review]

File: experimental/uefi/kernel/src/interrupts.rs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+lazy_static! {
+    static ref IDT: InterruptDescriptorTable = {
+        let mut idt = InterruptDescriptorTable::new();
+        idt.double_fault.set_handler_fn(double_fault_handler);
+        idt
+    };
+}
+
+extern "x86-interrupt" fn double_fault_handler(
```
Comment (rbehjati) [inline experimental/uefi/kernel/src/interrupts.rs:31]: What is "x86-interrupt"? Is this a type of linker that the Rust compiler already knows about or is it defined somewhere? Could you add a clarifying comment?

File: experimental/uefi/kernel/src/interrupts.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    stack_frame: InterruptStackFrame,
+    _error_code: u64,
+) -> ! {
+    // We're not calling `panic!` here because (a) the panic location would be all wrong (the error
+    // is not in the fault handler itself as the location would suggest), (b) in theory the
+    // panic handler itself could cause a double fault, and (c) we want to be sure that we shut
+    // down the machine after us.
```
Comment (rbehjati) [inline experimental/uefi/kernel/src/interrupts.rs:38]: nit: is "shut down the machine after us" an idiom that I don't know of? :)

**Review by rbehjati — APPROVED** [review]

File: experimental/uefi/kernel/src/interrupts.rs (hunk 1 above)
Comment (andrisaar) [reply experimental/uefi/kernel/src/interrupts.rs:31]: No, it's not about linker -- it changes the calling convention of the function and what code the compiler emits. The particular calling convention is required by the x86-64 ABI.

You can read more about the `x86-interrupt` in particular here: https://github.com/rust-lang/rust/issues/40180

In short, you don't really expect to see that used anywhere else except for interrupt handlers.

Comment (rbehjati) [reply experimental/uefi/kernel/src/interrupts.rs:31]: I think a comment with a link to that issue would help us understand what is going on later when we look at the code.

File: experimental/uefi/kernel/src/interrupts.rs (hunk 2 above)
Comment (rbehjati) [reply experimental/uefi/kernel/src/interrupts.rs:38]: Then I think describing it in those words would be clearer :)

**Pull request merged**